provider/provider.go

```diff
@@ -113,23 +140,32 @@ func (p *Provider) ConfigureProvider(_ context.Context, request *cqproto.Configu
 
 	p.extraFields = request.ExtraFields
 	p.dbURL = request.Connection.DSN
+
 	providerConfig := p.Config()
 	if err := defaults.Set(providerConfig); err != nil {
 		return &cqproto.ConfigureProviderResponse{
 			Diagnostics: diag.FromError(err, diag.INTERNAL),
 		}, nil
 	}
+
 	// if we received an empty config we notify in log and only use defaults.
 	if len(request.Config) == 0 {
```

Comment:
Looking good, some thoughts, tell me what you think about them:

why not let in the request the core passes the format of configuration? prior cores by default its set to HCL and used. new cores can decide based on configuration passed. 

Should the "provider" care / pass what format is supports? it just needs to know how to decode the block and pass as interface to main Configure function. I think also in the GetProviderConfig request we should do the same we can request yaml/hcl format if provider doesn't support yaml format we default back to hcl.